src/optimizer/utils.ts

```diff
@@ -164,6 +164,10 @@ function getTypesForPath(
     const instanceOfSD = fhirtypes.StructureDefinition.fromJSON(instanceOfDef);
     // NOTE: Normalize the path to remove indices and/or slice references
     const element = instanceOfSD.findElementByPath(path.replace(/\[[^\]]+\]/g, ''), fisher);
-    return element?.type;
+    if (element?.contentReference) {
+      return instanceOfSD.findElement(element.contentReference.slice(1))?.type;
```

Comment:
Since you are using `.slice(1)`, I think this assumes that the `contentReference` starts with a `#`.  But... I _think_ it is possible for a `contentReference` to start with a type id or URL (and _then_ `#element`).

Just to be safe, we should either:
* also test that the `contentReference` starts with `#`, OR
* do `.slice(element.contentReference.indexOf('#') + 1)`

I _think_ the 2nd one is safe, as I don't think contentReferences cross resources -- so the element id should exist, even if it references a different profile.  I'm not 100% sure, but I think it's a risk I'm willing to take.  That said, I'm fine w/ either of the above, so you can decide.